Remove Quirks::needsHDRPixelDepthQuirk()

annevk opened the pull request:
#### 95d938e7a998f0e15f36dcb56c1f15469025b70d
<pre>
Remove Quirks::needsHDRPixelDepthQuirk()
<a href="https://bugs.webkit.org/show_bug.cgi?id=254670">https://bugs.webkit.org/show_bug.cgi?id=254670</a>
rdar://107370756

Reviewed by Jer Noble and Brent Fulgham.

Remove a quirk YouTube no longer relies on per code inspection and at the same time make screen.colorDepth and pixelDepth share an implementation.

* Source/WebCore/loader/ResourceLoadStatistics.cpp:
(WebCore::encodeHashSet):
(WebCore::encodeOptionSet):
(WebCore::ResourceLoadStatistics::encode const):
(WebCore::decodeHashCountedSet):
(WebCore::decodeHashSet):
(WebCore::ResourceLoadStatistics::decode):
(WebCore::appendHashSet):
(WebCore::screenAPIEnumToString):
(WebCore::ResourceLoadStatistics::toString const):
(WebCore::ResourceLoadStatistics::merge):
* Source/WebCore/loader/ResourceLoadStatistics.h:

As a side effect, we no longer record pixelDepth separately from colorDepth.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableContentChangeObserver const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::allowLayeredFullscreenVideos const):
(WebCore::Quirks::shouldDisableLazyImageLoadingQuirk const):
(WebCore::Quirks::needsHDRPixelDepthQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::pixelDepth const): Deleted.
* Source/WebCore/page/Screen.h:
* Source/WebCore/page/Screen.idl:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/262330@main">https://commits.webkit.org/262330@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05ab2a004304adc929de6aca54a49d31ec5a0a3c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1218 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1252 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1291 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1987 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1088 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1203 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1303 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1321 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1251 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1228 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1138 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1130 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1851 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1163 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1121 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1105 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1113 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1152 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2211 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1161 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1110 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1097 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1131 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/315 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1188 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0f3e6e44dcf2e1505281a1fb0e0ecac83622c47b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/12119

brentfulgham commented on Source/WebCore/page/Screen.cpp:
```diff
@@ -98,21 +98,6 @@ unsigned Screen::colorDepth() const
-    RefPtr frame = this->frame();
-    if (!frame)
-        return 0;
-    if (frame->settings().webAPIStatisticsEnabled())
-        ResourceLoadObserver::shared().logScreenAPIAccessed(*frame->document(), ScreenAPIsAccessed::PixelDepth);
```
We do not need to keep this. It's part of a debugging mode for capturing Web API use; if we've replaced `pixelDepth` with `colorDepth`, we only care that the `colorDepth` is called. This is totally fine/safe to delete.

annevk replied:
Thanks for the quick response. To be perfectly clear, what this patch does is that if either the `pixelDepth` or `colorDepth` getter is invoked, it'll report `ScreenAPIsAccessed::ColorDepth`, due to this patch making them share an implementation.

If that's fine and it sounds like it is I'll do another pass here (probably tomorrow) to remove the corresponding `ScreenAPIsAccessed::PixelDepth` code.

brentfulgham replied:
Yes, this should be fine.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/05ab2a004304adc929de6aca54a49d31ec5a0a3c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/12119

annevk committed:
Remove Quirks::needsHDRPixelDepthQuirk()
https://bugs.webkit.org/show_bug.cgi?id=254670
rdar://107370756

Reviewed by Jer Noble and Brent Fulgham.

Remove a quirk YouTube no longer relies on per code inspection and at the same time make screen.colorDepth and pixelDepth share an implementation.

* Source/WebCore/loader/ResourceLoadStatistics.cpp:
(WebCore::encodeHashSet):
(WebCore::encodeOptionSet):
(WebCore::ResourceLoadStatistics::encode const):
(WebCore::decodeHashCountedSet):
(WebCore::decodeHashSet):
(WebCore::ResourceLoadStatistics::decode):
(WebCore::appendHashSet):
(WebCore::screenAPIEnumToString):
(WebCore::ResourceLoadStatistics::toString const):
(WebCore::ResourceLoadStatistics::merge):
* Source/WebCore/loader/ResourceLoadStatistics.h:

As a side effect, we no longer record pixelDepth separately from colorDepth.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableContentChangeObserver const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::allowLayeredFullscreenVideos const):
(WebCore::Quirks::shouldDisableLazyImageLoadingQuirk const):
(WebCore::Quirks::needsHDRPixelDepthQuirk const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::pixelDepth const): Deleted.
* Source/WebCore/page/Screen.h:
* Source/WebCore/page/Screen.idl:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/262330@main

webkit-commit-queue commented:
Committed 262330@main (95d938e7a998): <https://commits.webkit.org/262330@main>

Reviewed commits have been landed. Closing PR #12119 and removing active labels.
